python/tank_vendor/shotgun_authentication/defaults_manager.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+
+
+class DefaultsManager(object):
```

Comment:
See comments in the tk-desktop repos. I wonder if the default implementation here could almost be what we currently have in the desktop. That logic seems like a sensible out-of-the-box implementation for this and it would help simplify some things in desktop/desktop internal too.